omnigibson/systems/macro_particle_system.py

```diff
@@ -575,9 +652,10 @@ def get_particle_position_orientation(cls, idx):
         name = list(cls.particles.keys())[idx]
         # First, get local pose, scale it by the parent link's scale, and then convert into a matrix
         # Note that particles_local_mat already takes the parent scale into account when computing the transform!
-        parent_link = cls._particles_info[name]["link"]
+        parent_obj, parent_link = cls._particles_info[name]["obj"], cls._particles_info[name]["link"]
         local_mat = cls._particles_local_mat[name]
-        link_tf = T.pose2mat(parent_link.get_position_orientation())
+        link_tf = T.pose2mat(XFormPrim.get_local_pose(parent_obj)) if parent_link is None else \
+            T.pose2mat(parent_link.get_position_orientation())
```

Comment:
All of these `link is None` checks seem very brittle. Is it possible to explicitly store whether the current `group` is cloth versus rigid?

Reply:
try to be explicit with `if cloth: ... else: ...` rather than having inline `if else with link is None`. We will probably forget what `link is None` refers to after a few months.

Reply:
done!